example/test/function_component_test.dart

```diff
@@ -183,6 +183,134 @@ UseRefTestComponent(Map props) {
   ]);
 }
 
+final FancyInput = react.forwardRef((props, ref) {
+  final inputRef = useRef();
+
+  useImperativeHandle(
+    ref,
+    () {
+      print('FancyInput: useImperativeHandle re-assigns ref.current');
+      return {'focus': () => inputRef.current.focus()};
```

Comment:
This matches the JS implementation by using a Map, but I think it might be helpful to also show a Dart object being used (perhaps in another example), to illustrate that you can use typed objects and not just Maps.

```dart
class FancyInputApi {
  final void Function() focus;
  FancyInputApi._(this.focus);
}

final FancyInput = react.forwardRef((props, ref) {
  final inputRef = useRef();

  useImperativeHandle(
    ref,
    () {
      print('FancyInput: useImperativeHandle re-assigns ref.current');
      return FancyInputApi(() => inputRef.current.focus());
    },

    /// Because the return value of createHandle never changes, it is not necessary for ref.current
    /// to be re-set on each render so this dependency list is empty.
    [],
  );
});
```

It is admittedly a bit clunkier than using a Map, but it does provide static typing.

Thoughts?

Reply:
That makes sense!